bgzip.c

```diff
@@ -198,6 +203,9 @@ int main(int argc, char **argv)
 
         if ( index ) bgzf_index_build_init(fp);
         buffer = malloc(WINDOW_SIZE);
+#ifdef _WIN32
+        _setmode(f_src, O_BINARY);
+#endif
```

Comment:
If the decision here is to have bgzip open everything as binary (cf [second paragraph of this comment](https://github.com/samtools/htslib/pull/283#issuecomment-164023559)), wouldn't it be better to just use `hopen()` rather than sprinkling _WIN32 platform crap all over the code?